In [2]:
## First we will visualize the train data

In [3]:
import numpy as np
import pandas as pd
import sklearn

In [4]:
!ls -l china-real-estate-demand-prediction/train

total 4088
-rw-rw-r--@ 1 deepak  staff    6127  8 Aug 07:56 city_indexes.csv
-rw-rw-r--@ 1 deepak  staff  130589  8 Aug 07:56 city_search_index.csv
-rw-rw-r--@ 1 deepak  staff  174046  8 Aug 07:56 land_transactions_nearby_sectors.csv
-rw-rw-r--@ 1 deepak  staff  168460  8 Aug 07:56 land_transactions.csv
-rw-rw-r--@ 1 deepak  staff  590991  8 Aug 07:56 new_house_transactions_nearby_sectors.csv
-rw-rw-r--@ 1 deepak  staff  376737  8 Aug 07:56 new_house_transactions.csv
-rw-rw-r--@ 1 deepak  staff  322116  8 Aug 07:56 pre_owned_house_transactions_nearby_sectors.csv
-rw-rw-r--@ 1 deepak  staff  241849  8 Aug 07:56 pre_owned_house_transactions.csv
-rw-rw-r--@ 1 deepak  staff   63502  8 Aug 07:56 sector_POI.csv


In [35]:
land_transactions = pd.read_csv('china-real-estate-demand-prediction/train/land_transactions.csv')
city_search_index = pd.read_csv('china-real-estate-demand-prediction/train/city_search_index.csv')

land_transactions = pd.read_csv('china-real-estate-demand-prediction/train/land_transactions.csv')
land_transactions_nearby_sectors = pd.read_csv('china-real-estate-demand-prediction/train/land_transactions_nearby_sectors.csv')

new_house_transactions = pd.read_csv('china-real-estate-demand-prediction/train/new_house_transactions.csv')
new_house_transactions_nearby_sectors = pd.read_csv('china-real-estate-demand-prediction/train/new_house_transactions_nearby_sectors.csv')

pre_owned_house_transactions = pd.read_csv('china-real-estate-demand-prediction/train/pre_owned_house_transactions.csv')
pre_owned_house_transactions_nearby_sectors = pd.read_csv('china-real-estate-demand-prediction/train/pre_owned_house_transactions_nearby_sectors.csv')

sector_POI = pd.read_csv('china-real-estate-demand-prediction/train/sector_POI.csv')

#land_transactions.head(5)
#land_transactions.head()
#land_transactions_nearby_sectors.head()
#new_house_transactions.head()
#new_house_transactions_nearby_sectors.shape #head()
#pre_owned_house_transactions.head()
#pre_owned_house_transactions_nearby_sectors.head()
#sector_POI.tail()

month    sector  num_new_house_transactions  \
0  2019-Jan  sector 1                          52   
1  2019-Jan  sector 2                         145   
2  2019-Jan  sector 4                           6   
3  2019-Jan  sector 5                           2   
4  2019-Jan  sector 6                           5   

   area_new_house_transactions  price_new_house_transactions  \
0                         4906                         28184   
1                        15933                         17747   
2                          725                         28004   
3                          212                         37432   
4                          773                         15992   

   amount_new_house_transactions  area_per_unit_new_house_transactions  \
0                       13827.14                                    94   
1                       28277.73                                   110   
2                        1424.21                                   127   
3                         792.10                                   106   
4                         607.94                                    95   

   total_price_per_unit_new_house_transactions  \
0                                       265.91   
1                                       195.02   
2                                       356.05   
3                                       396.05   
4                                       151.99   

   num_new_house_available_for_sale  area_new_house_available_for_sale  \
0                             159.0                            15904.0   
1                            1491.0                           175113.0   
2                              40.0                             6826.0   
3                             161.0                            17173.0   
4                             189.0                            19696.0   

   period_new_house_sell_through  
0                           3.78  
1                          12.29  
2                           5.95  
3                          83.95  
4                          14.27

In [41]:

print(pre_owned_house_transactions_nearby_sectors.shape, " rows and columns.")
for cols in pre_owned_house_transactions_nearby_sectors:
    print(f"{cols} has {pre_owned_house_transactions_nearby_sectors[cols].isna().sum()} NaN values i.e. {100*pre_owned_house_transactions_nearby_sectors[cols].isna().sum()/pre_owned_house_transactions_nearby_sectors.shape[0]} % \n")

(5427, 6)  rows and columns.
month has 0 NaN values i.e. 0.0 % 

sector has 0 NaN values i.e. 0.0 % 

num_pre_owned_house_transactions_nearby_sectors has 0 NaN values i.e. 0.0 % 

area_pre_owned_house_transactions_nearby_sectors has 0 NaN values i.e. 0.0 % 

amount_pre_owned_house_transactions_nearby_sectors has 0 NaN values i.e. 0.0 % 

price_pre_owned_house_transactions_nearby_sectors has 0 NaN values i.e. 0.0 % 



## Input files analysis:
  <u> sector_POI </u> : 
  - It has 86 rows and 142 columns
  - 'surrounding_housing_average_price' has 4 NaN values (4.7%)
  - 'surrounding_shop_average_rent' has 3 NaN values (3.5%)

  <u> city_indexes </u> :
  - It had 7 rows and 74 columns.
  - total_fixed_assets_10k has 5 NaN values i.e. 71.4 %
  - main_business_taxes_and_surcharges_10k has 4 NaN values i.e. 57.1 %
  - total_fixed_asset_investment_10k has 7 NaN values i.e. 100.0 %
  - real_estate_development_investment_completed_10k has 4 NaN values i.e. 57.1 %
  - residential_development_investment_completed_10k has 4 NaN values i.e. 57.1 %
  - science_expenditure_10k has 1 NaN values i.e. 14.3 %
  - education_expenditure_10k has 1 NaN values i.e. 14.3 %

  <u> land_transactions </u> :
  - It has 5896 rows and 6 columns
  - A lot of 0.0 but no NaN

  <u> land_transactions_nearby_sectors </u> :
  - It has 5025 rows and 6 columns
  - A lot of 0.0 but no NaN

  <u> new_house_transactions </u> :
  - It has 5433 rows and 11 columns
  - num_new_house_available_for_sale has 14 NaN values i.e. 0.26 %
  - area_new_house_available_for_sale has 14 NaN values i.e. 0.26 %
  - period_new_house_sell_through has 14 NaN values i.e. 0.26 %

  <u> new_house_transactions_nearby_sectors </u> :
  - It has 5360 rows and 11 columns
  - No NaN values

  <u> pre_owned_house_transactions </u>
  - It has 5360 rows and 6 columns
  - price_pre_owned_house_transactions has 14 NaN values i.e. 0.3 %

  <u> pre_owned_house_transactions_nearby_sectors </u>
  - It has 5427 rows and 6 columns
  - No NaN values